pkg/route/validation/validation_test.go

```diff
@@ -1946,13 +1990,293 @@ func TestValidateTLS(t *testing.T) {
 					},
 				},
 			},
+
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid Reencrypt route as externalCertificate and certificate set",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationReencrypt,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "dummy",
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid Passthrough route as externalCertificate and certificate set",
```

Comment:
>passthrough route cannot have any certificate.

Agreed with that, but I think this test is meant to test false positives, to ensure it error'd out. I saw we have an existing test `Passthrough termination, cert` to test similar false positives.

Reply:
The problem with this test case is that it's same as `Passthrough termination, cert` which existed before your PR. The certificate field is checked before the externalCertificate and there is no dedicated `if` condition for both certificates set at the same time.

Reply:
The code will go through all the `if` conditions, no matter  `certificate` field is present or not, it will check `externalCertificate` 
So, for this test case `expectedErrors: 2` unlike `expectedErrors: 1` for `Passthrough termination, cert` test. 

I'm fine removing it though, it's just extra bit of testing, still relevant IMHO. wdyt ?

Reply:
I see what you mean. In that case why not setting all the fields which are checked (`CACertificate`, `Key` etc.)? `Certificate` field is no different from any of them. That may be misleading as the test name suggests like there is special handling when `Certificate` and `ExternalCertificate` are set like in the edge and reencrypt cases.

I think that more reasonable would be to test the setting of `ExternalCertificate` alone without `Certificate`because this was the new code introduced. Adding a test for all the `if`s of the passthrough route can be a bonus test case.

Reply:
Okay, that makes sense to make. I've removed this test since we are  checking `ExternalCertificate` alone without `Certificate`..And all  other `if`s are already checked individually.